common/src/main/java/haveno/common/UserThread.java

```diff
@@ -75,57 +82,68 @@ public static void await(Runnable command) {
-                throw new RuntimeException(e);
+                Thread.currentThread().interrupt(); // Restore interrupt status
+                log.error("Await was interrupted", e);
             }
         }
     }
 
     public static boolean isUserThread(Thread thread) {
         return thread == USER_THREAD;
-
     }
 
-    // Prefer FxTimer if a delay is needed in a JavaFx class (gui module)
     public static Timer runAfterRandomDelay(Runnable runnable, long minDelayInSec, long maxDelayInSec) {
-        return UserThread.runAfterRandomDelay(runnable, minDelayInSec, maxDelayInSec, TimeUnit.SECONDS);
+        return runAfterRandomDelay(runnable, minDelayInSec, maxDelayInSec, TimeUnit.SECONDS);
     }
 
-    @SuppressWarnings("WeakerAccess")
     public static Timer runAfterRandomDelay(Runnable runnable, long minDelay, long maxDelay, TimeUnit timeUnit) {
-        return UserThread.runAfter(runnable, new Random().nextInt((int) (maxDelay - minDelay)) + minDelay, timeUnit);
+        validateDelayParameters(minDelay, maxDelay);
+        long randomDelay = new Random().nextLong(maxDelay - minDelay) + minDelay;
+        return runAfter(runnable, randomDelay, timeUnit);
     }
 
     public static Timer runAfter(Runnable runnable, long delayInSec) {
-        return UserThread.runAfter(runnable, delayInSec, TimeUnit.SECONDS);
+        return runAfter(runnable, delayInSec, TimeUnit.SECONDS);
     }
 
     public static Timer runAfter(Runnable runnable, long delay, TimeUnit timeUnit) {
+        if (runnable == null || delay < 0 || timeUnit == null) {
+            throw new IllegalArgumentException("Runnable must not be null, delay must be non-negative, and timeUnit must not be null.");
+        }
         return getTimer().runLater(Duration.ofMillis(timeUnit.toMillis(delay)), () -> execute(runnable));
     }
 
     public static Timer runPeriodically(Runnable runnable, long intervalInSec) {
-        return UserThread.runPeriodically(runnable, intervalInSec, TimeUnit.SECONDS);
+        return runPeriodically(runnable, intervalInSec, TimeUnit.SECONDS);
     }
 
     public static Timer runPeriodically(Runnable runnable, long interval, TimeUnit timeUnit) {
+        if (runnable == null || interval < 0 || timeUnit == null) {
+            throw new IllegalArgumentException("Runnable must not be null, interval must be non-negative, and timeUnit must not be null.");
+        }
         return getTimer().runPeriodically(Duration.ofMillis(timeUnit.toMillis(interval)), () -> execute(runnable));
     }
 
     private static Timer getTimer() {
         try {
             return timerClass.getDeclaredConstructor().newInstance();
         } catch (InstantiationException | NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
-            String message = "Could not instantiate timer bsTimerClass=" + timerClass;
+            String message = "Could not instantiate timer timerClass=" + timerClass;
             log.error(message, e);
```

Comment:
You need to re-add throwing the exception: `throw new RuntimeException(message);`, else this is a compile-time error.